### PR TITLE
Safe script path

### PR DESCRIPTION
**Pull request by samwaseda**
Raised [here](https://github.com/pyiron/pyiron/issues/1261)

### CODE REVIEW

**Comment by pmrv** [comment]
This is not a specific issue with this PR, but in some of our `validate_ready_to_run` we raise `ValueError` and in some we raise `AssertionError`.  I'm not really partial either way, but I think we should agree on one type of exception to run or specify the semantics when to raise which.

**Comment by samwaseda** [comment]
According to [python doc](https://docs.python.org/3/library/exceptions.html):

- `ValueError`: Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError.
- `AssertionError`: Raised when an assert statement fails.

Since in this case nothing got an argument (let alone the right type), I would claim `ValueError` is not the right error. I'm gladly overruled if you disagree.

**Comment by liamhuber** [comment]
If you expect a path and get `None`, I think `TypeError` is best.

**Comment by samwaseda** [comment]
> If you expect a path and get `None`, I think `TypeError` is best.

I'm also fine with `TypeError`, but that sounds to me like the user actively put a wrong type, while in reality it's from developers' perspective that not setting the script path results in a different type than `str`.

A slightly different question: Would there be a practical problem if we set a potentially disputable warning (other than the fact that it could be misleading for the user)?

**Comment by liamhuber** [comment]
> I'm also fine with `TypeError`, but that sounds to me like the user actively put a wrong type, while in reality it's from developers' perspective that not setting the script path results in a different type than `str`.

`TypeError: ScriptJob.input.script_path [or whatever the arg name is] expects a path [or whatever it is you're expecting] but got None. Please provide a path before running.`

Anyone who reads this will understand why it's a `TypeError`, and the people who still don't get it, frankly, are probably not the type of people who read stack traces to begin with 😝 

> A slightly different question: Would there be a practical problem if we set a potentially disputable warning (other than the fact that it could be misleading for the user)?

Let's rather be technically correct -- the best type of correct there is 😉

**Comment by samwaseda** [comment]
Speaking of the distinction between `ValueError` and `TypeError`, there's quite a lot of work to be done, because as far as I know we mostly use `ValueError`, e.g. [here](https://github.com/pyiron/pyiron/blob/master/pyiron/atomistics/structure/atoms.py#L157), [here](https://github.com/pyiron/pyiron/blob/master/pyiron/atomistics/structure/atoms.py#L257), [here](https://github.com/pyiron/pyiron/blob/master/pyiron/atomistics/structure/atoms.py#L635), [here](https://github.com/pyiron/pyiron/blob/master/pyiron/atomistics/structure/atoms.py#L2336) or [here](https://github.com/pyiron/pyiron/blob/master/pyiron/atomistics/structure/atoms.py#L2392) (and I'm only naming those found in `atoms.py`)

**Comment by liamhuber** [comment]
Yeah, valid point. But this is definitely a niche enough issue that I'm content to "start doing it right now" rather than worry about perfect adherence throughout the codebase

File: tests/job/test_script.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    @classmethod
+    def setUpClass(cls):
+        cls.file_location = os.path.dirname(os.path.abspath(__file__)).replace(
+            "\\", "/"
+        )
+        cls.project = Project(os.path.join(cls.file_location, "test_sriptjob"))
+
+    @classmethod
+    def tearDownClass(cls):
+        file_location = os.path.dirname(os.path.abspath(__file__))
```
Comment (liamhuber) [inline tests/job/test_script.py]: Codacy appears correct to complain about this, indeed it doesn't appear to be used.

**Review by liamhuber — CHANGES_REQUESTED** [review]
Please fix the codacy "unused" complaint. Fixing the duplication complaint seems like overkill though -- IMO it's just fine to duplicate code *once*.

**Review by liamhuber — APPROVED** [review]

**Pull request merged**